HelpMyStreetFE/HelpMyStreetFE/Services/AddressService.cs

```diff
@@ -184,5 +190,48 @@ public async Task<double> GetDistanceBetweenPostcodes(string postCode1, string p
                 }
             }, $"{CACHE_KEY_PREFIX}-postcode-distances-{postCode1}-{postCode2}", RefreshBehaviour.DontWaitForFreshData, cancellationToken);
         }
+
+        public async Task<double> GetDistanceFromPostcodeForCurrentUser(string postCode, CancellationToken cancellationToken)
+        {
+            var httpContext = _httpContextAccessor.HttpContext;
+            if (httpContext.User != null && httpContext.User.Identity.IsAuthenticated)
+            {
+                var id = int.Parse(httpContext.User.FindFirst(ClaimTypes.NameIdentifier).Value);
+                var user =  await _userRepository.GetUser(id);
+                if (user != null && !String.IsNullOrEmpty("postCode"))
+                {
+                    return await GetDistanceBetweenPostcodes(user.PostalCode, postCode, cancellationToken);
+                }
+            }
+
+            return 0.0;
+        }
+
+        public async Task<LocationWithDistance> GetLocationWithDistance(IEnumerable<IContainsLocation> locationItem, CancellationToken cancellationToken)
+        {
+            return await GetLocationWithDistance(locationItem.First(), cancellationToken);
+        }
+
+        public async Task<LocationWithDistance> GetLocationWithDistance(IContainsLocation locationItem, CancellationToken cancellationToken)
+        {
+            var locationDetails = locationItem.GetLocationDetails();
+
+            if ((int)locationDetails.Location != 0)
+            {
+                locationDetails = await GetLocationDetails(locationDetails.Location, cancellationToken);
+            }
+
+            var lwd = new LocationWithDistance()
+            {
+                Distance = 0.0,
+                Location = locationDetails.Location,
+                LocationDetails = locationDetails
+            };
+
+            
+            lwd.Distance = await GetDistanceFromPostcodeForCurrentUser(locationDetails.Address.Postcode, cancellationToken);
```

Comment:
Cleaner to include this in the initial assignment to lwd above, rather than first putting Distance = 0.0 ?